ionoscloud/data_source_dataplatform_cluster.go

```diff
@@ -269,16 +309,21 @@ func setAdditionalDataplatformClusterData(ctx context.Context, d *schema.Resourc
 
 	/* get from api and set in schema the kubeconfig*/
 	if cluster.Id != nil {
-		kubeConfig, _, err := client.GetClusterKubeConfig(ctx, *cluster.Id)
+		kubeConfigMap, _, err := client.GetClusterKubeConfig(ctx, *cluster.Id)
 		if err != nil {
 			return fmt.Errorf("an error occurred while fetching the kubernetes config for cluster with ID %s: %w", *cluster.Id, err)
 		}
 
-		if err := d.Set("kube_config", kubeConfig); err != nil {
+		kubeConfig, err := json.Marshal(kubeConfigMap)
```

Comment:
_**NOTE:**_ The return type for `ClustersKubeconfigFindByClusterId` has changed and because of that `GetClusterKubeConfig` also returns a map instead of a string, this is why I'm making `json.Marshal` here.
_**BEFORE:**_

1. `ClustersKubeconfigFindByClusterId` -> `kube_config` string
2. Set `kube_config` schema attribute using the previously returned string
3. Unmarshal `kube_config` into a structure and populate `config` schema attribute using it.

_**AFTER:**_

1. `ClustersKubeconfigFindByClusterId` -> `kube_config` map
2. Marshal map into string
3. Set `kube_config` schema attribute with the previous string
5. Umarshal string into structure and populate `config` schema attribute using it

We should get rid of the `Unmarshal` method since we can operate on the returned map, but the code changes a little bit and this is not the scope of this PR so I did the `Marshal` in order to keep the current functionality.